### DISCUSSION TITLE
Is there a way to find all packages that are dependent to another package

### DISCUSSION THREAD

**Discussion by MscrmTools** [NuGet/Home#10825] [Q&A]
I would need to retrieve all packages that depend on another Nuget package. 

Is that achievable with Nuget.Protocol?

Thank you

**Comment by zivkan** [comment]
Not an easy way, no.

There's a sample in our docs on [how to get package metadata](https://docs.microsoft.com/en-us/nuget/reference/nuget-client-sdk#get-package-metadata). `IPackageSearchMetadata` has a list of `PackageDependencyGroup`, each of which represent a TFM, and they have the list of packages.

You'd need to iterate all the packages. On nuget.org, you can use the [catalog resource](https://docs.microsoft.com/en-us/nuget/api/catalog-resource) to determine every package ever pushed. You'll need to deduplicate the package version list, exclude deleted packages, and optionally exclude unlisted packages. There's [a sample catalog reader](https://github.com/NuGet/Samples/tree/main/CatalogReaderExample) available to help get started if you want to go down this route.

**Reply by MscrmTools** [reply]
Thank you. I will look at these docs
